Packs/Uptycs/Integrations/Uptycs/Uptycs.py

```diff
@@ -2199,17 +2249,131 @@ def uptycs_post_saved_query_command():
+                              'x-amz-server-side-encryption-customer-key',
+                              'x-amz-server-side-encryption-customer-key-md5',
+                              'x-requested-with']
+    headers['Access-Control-Request-Headers'] = ','.join(access_control_headers)
+    _, query_string = url.split('?')
+    params = dict(param.split('=') for param in query_string.split('&'))
+
+    keys = ['x-amz-server-side-encryption-customer-key',
+            'x-amz-server-side-encryption-customer-key-MD5']
+
+    for key in keys:
+        value = params.get(key)
+        if value is not None:
+            headers[key] = unquote(value)
+
+    return headers
+
+
+def uptycs_get_carves_file_command():
+    query_results = uptycs_get_carves_link_source()
+    url = query_results.get('url')
+    if url is None:
+        return_error("Error in retrieving file url")
+
+    carve_id = demisto.args().get('carve_id')
+    fileheaders = get_carve_headers(url)
+    response = requests.get(url, headers=fileheaders)
+    filename = carve_id + '.tar'
+    if response.status_code == 200:
+        return demisto.results(fileResult(filename, response.content))
+    return_error('Error , file fetching failed , got response ' + str(response.status_code))
+
+
+def uptycs_get_carves():
+    """
+    return list of carves in Uptycs
+    """
+    http_method = 'get'
+    api_call = '/carves'
+
+    path = demisto.args().get('path')
+    if path is not None:
+        api_call = '%s?filters={"path":{"equals":"%s"}}' %\
+            (api_call, path)
+
+    return restcall(http_method, api_call)
+
+
+def uptycs_get_carves_source_command():
+
+    query_results = uptycs_get_carves()
+    context = query_results.get('items')
+    context_entries_to_keep = ['id', 'assetId', 'path', 'createdAt',
+                               'updatedAt', 'status', 'deletedUserName',
+                               'deletedAt', 'assetHostName', 'offset',
+                               'length']
+
+    if context is not None:
+        remove_context_entries(context, context_entries_to_keep)
+
+    human_readable = tableToMarkdown('Uptycs Carves',
+                                     context, context_entries_to_keep)
+
+    entry = {
+        'ContentsFormat': formats['json'],
+        'Type': entryTypes['note'],
+        'Contents': query_results,
+        'HumanReadable': human_readable,
+        'EntryContext': {
+            'Uptycs.Carves(val.id == obj.id)': context
+        }
+    }
+    return entry
+
+
 def uptycs_test_module():
     """check whether Uptycs API responds correctly"""
     http_method = 'get'
     api_call = '/assets?limit=1'
 
     query_results = restcall(http_method, api_call)
 
-    if query_results == 0:
-        return False
-    else:
+    if query_results.get('items') is not None:
+        if len(query_results.get('items')) == 0:
+            return False
         return True
+    else:
+        return False
```

Comment:
If a command fail (in this case the test_module), it should raise an exception with an informative message with the failing reason. Please update this.